renderer/components/home/running.js

```diff
@@ -236,63 +147,210 @@ const ContentContainer = styled.div`
   z-index: 10;
 `
 
-const Running = ({
-  testGroupName,
-  progressLine,
-  percent,
-  runningTestName,
-  logLines,
-  eta,
-  error,
-  onKill,
-  stopping
-}) => {
+const Title = ({ groupName }) => (
+  <Heading h={2}>{groupName}</Heading>
+)
 
+Title.propTypes = {
+  groupName: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.instanceOf(FormattedMessage)
+  ])
+}
+
+const MemoizedTitle = React.memo(Title)
+
+const Running = ({ testGroupName }) => {
   const [logOpen, setLogOpen] = useState(false)
+  const [error, setError] = useState(null)
+  const [runningTestName, setRunningTestName] = useState(null)
+  const [progressLine, setProgressLine] = useState('')
+  const [logLines, setLogLines] = useState([])
+  const [percent, setPercent] = useState(0)
+  const [eta, setEta] = useState(-1)
+  const [stopped, setStopped] = useState(false)
+  const [isStopping, setIsStopping] = useState(false)
+
+  let runner = null
+
+  useEffect(() => {
+    const { ipcRenderer, remote } = require('electron')
+    ipcRenderer.on('ooni', onMessage)
+
+    debug('running', testGroupName)
+
+    const Runner = remote.require('./utils/ooni/run').Runner
+    runner = new Runner({ testGroupName })
+    runner
+      .run()
+      .then(() => {
+        setStopped(true)
+        Router.push('/test-results')
+      })
+      .catch(error => {
+        debug('error', error)
+        Raven.captureException(error, {
+          extra: { scope: 'renderer.runTest' }
+        })
+        setError(error)
+      })
+
+
+    return () => {
+      const { ipcRenderer } = require('electron')
```

Comment:
```suggestion
```

Why do you import `ipcRenderer` again in this inner scope? Isn't `ipcRenderer` already part of the closure of this function because you already import it above?

Reply:
removed